[FAB-18188] Log orderer and peer cert expiration date upon startup

denyeart opened the pull request:
FAB-17000 added a warning if expiration is within a week.

This change additionally info logs the future enrollment, tls server, and tls client
cert expiration dates upon startup, regardless of whether they will expire within a week or not.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

GrapeBaBa commented:
How to re-trigger the travis build?

JonathanLevi commented:
reverify Travis

JonathanLevi commented:
So 'reverify Travis' doesn't work, though 'reverify jenkins'' used to.

The simplest (old fashioned, but working) is to push something to the same branch... (Can be more than one commit at a time, in case you want to do and undo a change, or so)

GrapeBaBa commented:
@JonathanLevi For this PR currently I have no more code need to push, unit test passed in my local, I just want to trigger the build again.

JonathanLevi commented:
BTW: I am not in front of a PC/Mac and w/o my glasses ;-), but given #1803 (and #1809 ), the build is probably not broken due to your current changes. I believe this will be resolved tomorrow, which will affect/fix several pending PRs....

Let's give it a day?

GrapeBaBa commented:
Sure, enjoy your weekend.

kchristidis commented on docs/dev-setup/ca-setup.md:
```diff
@@ -35,10 +35,10 @@ Furthermore, logging levels can be enabled/disabled by adjusting the following s
 
 Alternatively, these fields can be set via environment variables, which---if set---have precedence over entries in the yaml file.  The correspnding environment variables are named as follows:
```
s/correspnding/corresponding

bramwelt commented:
reverify jenkins

bramwelt commented:
reverify jenkins

JonathanLevi commented:
Good to know @christo4ferris, thanks.

@mastersingh24: can we have your 'blessing' on this? (we initially talked about "CA_..." then on #fabric-dev it was suggested to use "HLF_CA_"... so that the "CORE_" will later change to "HLF_CORE_").

Either way, it's good to update the "OBC" prefix for starters, and remove the 2 not-anymore-necessary `ca/util.go` helper functions...

JonathanLevi commented:
Good to know @christo4ferris, thanks. Suggested to add these to the Wiki in #1817/#1816.

@mastersingh24: can we have your 'blessing' on this? (we initially talked about "CA_..." then on #fabric-dev it was suggested to use "HLF_CA_"... so that the "CORE_" will later change to "HLF_CORE_").

Either way, it's good to update the "OBC" prefix for starters, and remove the 2 not-anymore-necessary `ca/util.go` helper functions...

JonathanLevi commented:
Looks good to me (see also notes on the #1810). @christo4ferris, @srderson @ghaskins: mind merging this too (regardless of the known #680 failure)?

Not a biggy, but helps getting rid of more redundant functions (in `membersrvc/ca/util.go`) and the old OBC... refs.

srderson commented:
are we waiting for comment from @mastersingh24?

JonathanLevi commented:
@srderson Not really/anymore.

We had a discussion on #fabric-dev, and I discussed it with @mastersingh24 (after asking for his blessings above), considering everything, @mastersingh24/Gari suggested that we are good to go with the  `MEMBERSRVC_CA_` prefix instead of the original `HLF_CA_` (to which @ghaskins agreed as well).

mastersingh24 commented:
I am good

srderson commented:
LGTM. Thanks @GrapeBaBa

denyeart committed:
[FAB-18188] Log orderer and peer cert expiration date upon startup

FAB-17000 added a warning if expiration is within a week.

This change additionally info logs the future enrollment, tls server, and tls client
cert expiration dates upon startup, regardless of whether they will expire within a week or not.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
